Lib/idlelib/idle_test/test_calltip.py

```diff
@@ -48,6 +49,7 @@ class Get_argspecTest(unittest.TestCase):
     # but a red buildbot is better than a user crash (as has happened).
     # For a simple mismatch, change the expected output to the actual.
 
+    @requires_docstrings
```

Comment:
Why this is needed? There are checks like `if List.__doc__ is not None` in the code.

Reply:
Someone, likely me, must have tested with `-OO` several years ago and added the checks where needed to pass They were not needed for builtins list, list.append, re.sub, and types.MethodType.

I had no idea about compiling without docstrings.  The following suggests that this in only an option on non-Windows
```

test.support.MISSING_C_DOCSTRINGS
    Return True if running on CPython, not on Windows, and configuration not set with WITH_DOC_STRINGS.
```
True?  In any case, I am unsure about the interaction between compiling without docstrings, running with -OO, HAVE_DOCSTRINGS and the decorator.  So I am not sure of the correct fix.

It appears that compiling without docstrings does not affect python function docstrings.  test_properly formatted only has a python docstring, and this did not fail in Zackery's test.  Without a guard, it does fail when I tested with -OO.  It must be newer than when guards were added.  Also, other idle test modules now fail with -OO.  This suggests that separating tests of builtin and python docstrings would allow tests of either to run if only the other are missing.  (It also suggests the need for another issue re tests that fail with -OO.)